cli/src/Morphir/Web/DevelopApp.elm

```diff
@@ -1788,7 +1861,16 @@ httpMakeModel =
                             HttpError httpError
 
                         Ok result ->
-                            ServerGetIRResponse result
+                            case Repo.fromDistribution result of
+                                Ok repo ->
+                                    ServerGetIRResponse ( result, repo )
+
+                                Err error ->
+                                    let
+                                        _ =
+                                            Debug.log "e" error
+                                    in
+                                    HttpError (Http.BadBody "Could not transform Distribution to Repo")
```

Comment:
You should move this bit of code into the `update` function. Generating an HTTP error doesn't seem to be right. The `update` function is the one that should create the repo and update the model to put the application in an errored state if the operation fails. In general any substantial processing should happen within the update function.